### ISSUE TITLE
[REVIEW]: NetworkChange: Analyzing Network Changes in R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2708]
**Submitting author:** @ysohn (<a href="http://orcid.org/0000-0002-9496-1907">Yunkyu Sohn</a>)
**Repository:** https://github.com/jongheepark/NetworkChange
**Branch with paper.md** (empty if default branch): 
**Version:** v0.6
**Editor:** @arfon
**Reviewers:** @akbaritabar, @marcjwilliams1, @martinmodrak
**Archive:** Pending

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6/status.svg)](https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6"><img src="https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6/status.svg)](https://joss.theoj.org/papers/cb2034afbb7ea28d8c94bc4b421034b6)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/jongheepark/NetworkChange" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@akbaritabar & @marcjwilliams1 & @martinmodrak, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kakiac know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jongheepark/NetworkChange">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ysohn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jongheepark/NetworkChange">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ysohn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jongheepark/NetworkChange">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@ysohn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @akbaritabar, @marcjwilliams1, @martinmodrak it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.4135/9781604265781 is OK

MISSING DOIs

- 10.18637/jss.v024.i05 may be a valid DOI for title: Fitting position latent cluster models for social networks with latentnet
- 10.1214/19-ba1147 may be a valid DOI for title: Detecting Structural Changes in Longitudinal Network Data
- 10.1214/15-aoas839 may be a valid DOI for title: Multilinear tensor regression for longitudinal relational data
- 10.21236/ada458734 may be a valid DOI for title: Latent space approaches to social network analysis

INVALID DOIs

- None
```

**Comment by kakiac** [comment]
Editorial Comments carried over from [Pre-review] https://github.com/openjournals/joss-reviews/issues/2471 (@kakiac to check):

-[ ] Your software repository does not appear to include an open source license
-[ ] Your paper is missing an explicit Statement of Need section, which is now required for articles as described in our documentation
-[ ] Please remove these lines from your paper's YAML header: aas-doi: 10.3847/xxxxx and aas-journal: Journal of Open Source Software.
-[ ] Also, overall your paper is quite short. Although JOSS papers are generally only one or two pages on average (250-1000 words), your paper just barely passes that mark. I would recommend adding a bit more content, perhaps an example or more explanation/details of how the package works.
-[ ] It looks like a few of your references may be missing DOIs

**Comment by kakiac** [comment]
👋🏼 @ysohn @akbaritabar, @marcjwilliams1, @martinmodrak this is the review thread for the paper. All of our communications will happen here from now on.

All reviewers should have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#2708` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 6 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@kakiac) if you have any questions/concerns.

**Comment by ysohn** [comment]
@kakiac We understand. Thank you very much! Also we deeply appreciate for the reviewers' time and efforts.

**Comment by arfon** [comment]
@akbaritabar, @marcjwilliams1, @martinmodrak - how are you getting on with your reviews here? Please note we try to _complete_ the review process in 6 weeks which means that it's helpful for you all to provide initial feedback to the authors earlier than this.

**Comment by marcjwilliams1** [comment]
I don't seem to be able to check off items on the checklist, if I click through to https://github.com/openjournals/joss-reviews/invitations as suggested I get the following message
`Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account.`

**Comment by arfon** [comment]
@whedon re-invite @marcjwilliams1 as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@marcjwilliams1 please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
@marcjwilliams1 - can you try clicking the invite link again now? I believe the invites expire after a week.

**Comment by marcjwilliams1** [comment]
@arfon working now, thanks.

**Comment by akbaritabar** [comment]
## Letter to authors
Thank you for your trust and allowing me to take part in this interesting experience as I am studying peer review as a subject of research and the way JOSS is handling it was really interesting to me.

I must start by saying that I am a social scientist who uses R, Python and network analysis tools and techniques on daily basis. That means I have evaluated the submission with the above background and most of my points below are not technical at all. Reading the package paper, vignette and the authors' 2020 paper in Bayesian Analysis journal, I was ready to give up on the review and state a lack of _technical_ confidence. Then I decided to act as an _advanced beginner_ user of the package and try to provide feedback on my experience that _might_ be useful for authors and the editor (I certainly hope so). I will state some main points that I faced and below some minor suggestions. Further below I have copied some of the points from the review checklist and added comments on them that I needed to describe.

A bit more description about the empirical network data used would be nice (e.g., I found this in the 2020 paper by the authors in Bayesian Analysis journal which made it much clearer for me (quoting this from that paper's text) "The structure of military alliance networks reflects the distribution of military power among coalitions of states, which changes over time in response to exogenous shocks to the international system or endogenous network dynamics. However, there has been no study that investigates changes in coalition structures of military alliance networks over time using a principled statistical approach. A main reason is the lack of statistical methods that model unknown structural changes in longitudinal network data.").

Largest possible size of the network and longitudinal waves that can be analyzed and visualized by the package is not mentioned in paper or vignette. For the current example in vignette, there was the "BreakDiagnostic" step that on my laptop (core i 7, 16GB RAM) took a few minutes to be done, I was wondering if authors can give estimates of possible time it can take and what factors can affect that in network attributes and analysis terms for users to have an idea.

An example of how objects from popular R libraries for network analysis (e.g., igraph and statnet suite) or external formats (e.g., Pajek plain text format) can be converted and used with NetworkChange will be really helpful. At first it feels like array of temporal snapshots is easy to build, but it might not be for all potential users. I took the time to test it on a fake co-authorship network and it was not as straightforward as I hoped it would be (please see above in first paragraph that I am an advanced beginner and it might only be my problem). I used a bipartite edgelist saved in CSV including source/target columns for authors and publication nodes and a publication year column (see below for the example). I constructed a bipartite graph with igraph, then induced yearly subgraphs (to have temporal dimension to explore with NetworkChange) and then projected them to one-mode (authors). I then used array from R to get close to the MajourAlly shape, but I failed and couldn't understand how I need to provide the dimensions to then use NetworkChange on the sample. Providing a bit of description on this would allow the package to be used by a wider range of audience.

This is example of bipartite ties I tried to convert and used and failed in the latest step:

```R

# example data (edges table) in csv format to copy/paste

source,target,PUBYEAR
author_1,paper_2,2007
author_2,paper_2,2007
author_3,paper_2,2007
author_4,paper_2,2007
author_5,paper_2,2007
author_1,paper_3,2011
author_2,paper_3,2011
author_3,paper_3,2011
author_4,paper_3,2011
author_5,paper_3,2011
author_1,paper_4,2012
author_2,paper_4,2012
author_3,paper_4,2012
author_4,paper_4,2012
author_5,paper_4,2012
author_1,paper_1,2013
author_2,paper_1,2013
author_3,paper_1,2013
author_4,paper_1,2013
author_5,paper_1,2013


# example data (vertices table) in csv format to copy/paste

vertices,type
author_1,TRUE
author_2,TRUE
author_3,TRUE
author_4,TRUE
author_5,TRUE
paper_1,FALSE
paper_2,FALSE
paper_3,FALSE
paper_4,FALSE


# Preparing to use with NetworkChange
require(tidyverse)
require(igraph)

edges_table <- read_csv('./edges.csv') %>% 
  arrange(PUBYEAR)

vertices_table <- read_csv('./vertices.csv')

# build a bipartite paper-author graph
g <- graph_from_data_frame(d = edges_table, vertices = vertices_table, directed = F)

yearly_g <- list()
for (year in unique(edges_table$PUBYEAR)) {
  # delete.vertices = FALSE because we need same dim for all matrices
  g_year <- subgraph.edges(g, E(g)[PUBYEAR == year], delete.vertices = FALSE)
  g_aut <- bipartite.projection(g_year, which = 'true')
  g_adj <- as.matrix(get.adjacency(g_aut, edges = F, names = T))
  yearly_g[[paste0('y_', as.character(year))]] <- g_adj
}

# Here I am not sure how to give the right dimensions and fail in building the right array that NetworkChange will use
arr_yearly_g <- array(yearly_g)

### the error I get is like this:

# use networkchange
G <- 100
set.seed(1990)
test.run <- NetworkStatic(arr_yearly_g, R=2, mcmc=G, burnin=G, verbose=0,
                          v0=10, v1=4*2)

# Error in array(Y, dim = c(dim(Y)[1], dim(Y)[2], 1)) : 
#   negative length vectors are not allowed

```

Below I present some minor points as suggestions to be revised in text. I am sorry that I don't have much technical suggestions to offer and I hope other reviewers and editor would provide enough comments for those aspects.

### Minor points
#### In JOSS paper
- "The complete guide for using core functions of the package is presented at https://github.com/jongheepark/NetworkChange as its vignette with an empir- ical data set analysis example." -> This is the link to repository not the vignette (https://cran.r-project.org/web/packages/NetworkChange/vignettes/NetworkChange.html)
- In both paper and the vignette (see below), figure 1 ("Summary of selected features and functions of the package.") is used but in text there no description of it. It is a nice figure summarizing the main features and function names, it would be nicer to have a few sentences describe it (IMO).
- "library(NetworkChange)" is missing from the first code chunk of the paper (it is present in the vignette).
- "We follow the COW dataset’s coding of “major powers” (the United Kingdom, Germany, Austria-Hungary, France, Italy, Russia, the United States, Japan, and China) in the analysis. We aggregated every 2 year network from the original annual binary networks to increase the density of each layer." -> in the script accompanying this part you are dropping CHN and USA since they don't have ties to other nodes, but in text it is not mentioned that they are dropped.
- - "Users can choose the number of clusters in each regime by ‘n.cluster}" -> "` `" instead of "}"

#### In Vignette
- "Users can change plot settings by changing options in `ggnet`." -> maybe, but changing options which _are passed to_ `ggnet`?
- "In this section, We estimate the chagepoints of" -> changepoints
- "We also highlight the latent cluster structure using the k-means clustering method over the estimated latent node positions of each regime. drawPostAnalysis() provides an option to include k-means clustering results of the latent node positions for each regime." -> I felt the two sentences were similar.
- "In this section, we analyze changes in the international military alliance network among major powers. The data set is originally from (???)" -> reference in parenthesis is missing from the vignette.
- "Our goal in this section is to detect structural changes in the longitudinal alliance network among major powers using HNC. We follow the COW dataset’s coding of “major powers” (the United Kingdom, Germany, Austria-Hungary, France, Italy, Russia, the United States, Japan, and China) in the analysis. We aggregated every 2 year network from the original annual binary networks to increase the density of each layer." -> in the script accompanying this part you are dropping CHN and USA since they don't have ties to other nodes, but in text it is not mentioned that they are dropped.
- "Users can choose the number of clusters in each regime by ‘n.cluster}" -> "` `" instead of "}"
- Figure captions present in the paper are missing from the vignette, but I am not sure if the vignette would support them or not? If it does, it would be nice to have them here as well.
- "International military alliances, 1648-2008. (2009). Washington, DC: CQ Press. doi:10.4135/ 9781604265781" -> Reference to empirical data used is missing from vignette

#### In package documentation PDF from CRAN
- "orthgonalization" -> "orthogonalization"

#### Points from Review checklist that I needed to elaborate
- General checks
    - [x] Contribution and authorship: Has the submitting author (@ysohn) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
        - The submitting author seems to have joined recently (https://github.com/jongheepark/NetworkChange/graphs/contributors) but the first author have worked on this repository from late 2017. The current order of authorship seems to respect that IMO.
    - [x] Substantial scholarly effort: Does this submission meet the scope eligibility described in the JOSS guidelines
        - There are two peer reviewed publications (2017 and 2020) based on the methodology as far as I can see and the lead author is the maintainer of CRAN task view on Bayesian analysis, I feel confident the contribution would be good. But I might not be technically qualified to judge that.
- Functionality
    - [x] Installation: Does installation proceed as outlined in the documentation?
        - Installation from CRAN and latest version from Github went fine on Mac OS.
        - But, once installed from github, the vignette of package is not found in R with ` vignette(package = "NetworkChange") no vignettes found `
- Documentation
    - [ ] Automated tests: Are there automated tests or manual steps described so that the functionality of the software can be verified?
        - Vignette and paper gives empirical example and scripts to see how functions work. But I am not sure if I can find automated tests.
    - [ ] Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
        - On the package repository or its Wiki there is no guide on this, but Github issue tracker and pull requests should be the intended ones, I suppose.
- Software paper
    - [x] Summary: Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
        - The current description can be further improved. For example, for me as someone who knows about network analysis, block-modeling, community detection and similar techniques, I cannot say the whole methodology and descriptions are clear but that could be my lack of enough background knowledge.
    - [ ] Quality of writing: Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
        - I have given minor suggestions for paper and the vignette below.

**Comment by marcjwilliams1** [comment]
Overall everything looks fine, I get the same outputs when I run the vignette but would be nice to see some automated tests jongheepark/NetworkChange#3 . A note on community guidelines should also be added somewhere jongheepark/NetworkChange#4 .

**Comment by kakiac** [comment]
👋🏼  @akbaritabar, @marcjwilliams1 many thanks for your reviews and comprehensive comments.

@ysohn can you please start addressing the reviewers' comments? I will be creating issues in your repo from @akbaritabar's comments to help us keep track of the comments to be addressed 😄 

@martinmodrak I see that you have not checked any of your checklist items - can you edit it or are you having similar problems like @marcjwilliams1 ?

**Comment by kakiac** [comment]
@whedon re-invite @martinmodrak as reviewer

**Comment by whedon** [comment]
@martinmodrak already has access.

**Comment by ysohn** [comment]
@akbaritabar, @marcjwilliams1 Thank you so much for your reviews!

@kakiac We will address the issues shortly and commit on github starting next week. Thank you!

**Comment by martinmodrak** [comment]
Sorry for not being active, some life stuff was happening. I'll have my review ready by Monday.

**Comment by martinmodrak** [comment]
So I did a first round of review. I think I can complement @akbaritabar in that I've never done network analysis, but I have background in Bayesian inference and even some experience with fitting HMMs. I however have to be clear that my exposure to Bayesian methods is primarily via the Stan language and the associated community (e.g. Gelman, Vehtari, Betancourt, Simpson) which tends to have opinionated worldview on how Bayesian statistics should be done which may not necessarily match the consensus of the whole field. So feel free to challenge specific propositions/suggestions I make.

The problem attacked by the package is hard. My first impression upon reading the associated Bayesian analysis paper was "I wonder if this kind of model can be fit reliably with any available method.". To some extent my suspicion has been vindicated - I've noticed some problems that threaten the validity of the computation (https://github.com/jongheepark/NetworkChange/issues/6), so this is currently my biggest concern. In the case I am not wrong about the issues (which I might be), resolving them completely might turn out to be a significant research project on its own and I don't want to force the authors to do that. I would generally be happy if the package provides diagnostics and warnings to the user in case something looks fishy.

The paper itself (and the package) relies heavily on the Bayesian analysis paper for any deeper explanations. I am mostly OK with this, although I can see how adding at least rough description of the actual model to the JOSS paper would be beneficial. I will be happy if @kakiac as the editor weighs in if this is appropriate.

**Comment by martinmodrak** [comment]
I should also add that installation was easy and following the examples was also easy, so definitely good job there.

**Comment by kakiac** [comment]
@whedon check references

**Comment by kakiac** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.4135/9781604265781 is OK

MISSING DOIs

- 10.18637/jss.v024.i05 may be a valid DOI for title: Fitting position latent cluster models for social networks with latentnet
- 10.1214/19-ba1147 may be a valid DOI for title: Detecting Structural Changes in Longitudinal Network Data
- 10.1214/15-aoas839 may be a valid DOI for title: Multilinear tensor regression for longitudinal relational data
- 10.21236/ada458734 may be a valid DOI for title: Latent space approaches to social network analysis

INVALID DOIs

- None
```

**Comment by kakiac** [comment]
👋  @ysohn, hope you are well, I have tried to organise the reviewers' comments up to now in separate issues in your repository to help us all track progress :). If you hover over them, you can see their status.

Hope it helps!

Many thanks to all reviewers for the comments up to now, @ysohn let us know if you need any help or further clarification on any of our comments.

Comments from **JOSS editors**:

- https://github.com/jongheepark/NetworkChange/pull/1
- https://github.com/jongheepark/NetworkChange/issues/7

Comments from **@marcjwilliams1**:
- https://github.com/jongheepark/NetworkChange/issues/3
- https://github.com/jongheepark/NetworkChange/issues/4

Comments from **@martinmodrak**:
- https://github.com/jongheepark/NetworkChange/issues/5
- https://github.com/jongheepark/NetworkChange/issues/6

Comments from **@akbaritabar**:
- https://github.com/jongheepark/NetworkChange/issues/8
- https://github.com/jongheepark/NetworkChange/issues/9
- https://github.com/jongheepark/NetworkChange/issues/10
- https://github.com/jongheepark/NetworkChange/issues/11
- https://github.com/jongheepark/NetworkChange/issues/12
- https://github.com/jongheepark/NetworkChange/issues/13
- https://github.com/jongheepark/NetworkChange/issues/14

**Comment by kakiac** [comment]
> Sorry for not being active, some life stuff was happening. I'll have my review ready by Monday.

No problem @martinmodrak, it's challenging times, we appreciate your help with this :)

**Comment by ysohn** [comment]
@martinmodrak No problem. Wish you the best for the issues.. We deeply appreciate your comments. 

> Sorry for not being active, some life stuff was happening. I'll have my review ready by Monday.

**Comment by ysohn** [comment]
@kakiac Thank you very much for compiling the issues! 

> 👋 @ysohn, hope you are well, I have tried to organise the reviewers' comments up to now in separate issues in your repository to help us all track progress :). If you hover over them, you can see their status.

**Comment by ysohn** [comment]
@akbaritabar, @marcjwilliams1, @martinmodrak, @kakiac 

Dear Reviewers and Editor,

We deeply appreciate your detailed reviews and suggestions. It looks like some of the issues may take time for us to resolve. We will try to complete addressing the issues by January. 

Wish all of you stay healthy, and have warm holidays!

All my best

**Comment by kakiac** [comment]
@yson no problem at all. I will set an automatic reminder to all of us for mid January - in the meantime hope you all take some time to relax and "recharge" your batteries :)

**Comment by kakiac** [comment]
@whedon remind @yson @kakiac in six weeks

**Comment by whedon** [comment]
@yson @kakiac doesn't seem to be a reviewer or author for this submission.

**Comment by kakiac** [comment]
@whedon remind @yson in six weeks

**Comment by whedon** [comment]
@yson doesn't seem to be a reviewer or author for this submission.

**Comment by ysohn** [comment]
> @yson no problem at all. I will set an automatic reminder to all of us for mid January - in the meantime hope you all take some time to relax and "recharge" your batteries :)

Thank you! Hope all of you have great holidays and a happy new year.

**Comment by kakiac** [comment]
@whedon remind @ysohn in six weeks

**Comment by whedon** [comment]
Reminder set for @ysohn in six weeks

**Comment by whedon** [comment]
:wave: @ysohn, please update us on how things are progressing here.

**Comment by ysohn** [comment]
@kakiac Dear editor, Happy new year. We are working to resolve the multimodality problem raised by one of the reviewers. It takes longer than expected as we have to experiment several candidates for identification of the MCMC samples. It'd be great if we could have more time to revise the package.

**Comment by martinmodrak** [comment]
@ysohn: if that's in response to my review,  I reiterate that I would find it sufficient for publication in JOSS if the package provided diagnostics and clearly documented how a user can verify that the results can be trusted - or which parts of the results should be taken seriously (e.g. if the `U` parameters are not interpretable). My hunch is that for general datasets, the identification issues are irresolvable, so maybe good diagnostics to check if the model works for a given dataset is the best we can hope for. Obviously, if you can resolve or mitigate the issues,  its good for everyone,  but I didn't intend to send you on such a large project.

**Comment by ysohn** [comment]
@martinmodrak  Thank you so much for your reply. I understand your point now. Without artificial constraints, factorized estimates in general have degrees of freedom coming from rotation and scaling. In our model, the sum of outer products of the estimates are meant to produce matrices that are eventually used in the model whereas the factorized estimates themselves are instrumental for this recovery. Thus, if we do not impose artificial constraints for the factorized estimates, the key for the diagnosis is to show the convergence up to the resulting matrices. We will work on this. Thank you for your patience.

**Comment by arfon** [comment]
:wave: @ysohn - how are you getting along here? Do you think you might be able to make your updates soon?

**Comment by arfon** [comment]
:+1: no problem. Thanks for the update @ysohn.

**Comment by ysohn** [comment]
@arfon Thank you for your understanding.

**Comment by arfon** [comment]
@whedon assign me as editor

**Comment by danielskatz** [comment]
@arfon - This seems stuck?

**Comment by arfon** [comment]
:wave: @ysohn – do you think you might be able to complete your updates in the next month? If not, I'm going to recommend we close this review and encourage a resubmission in the future if/when you are able to make the changes requested by our reviewers.

**Comment by ysohn** [comment]
@arfon I deeply apologize for the delay. We agree to close the review if we do not complete the changes requested by the reviewers by the end of November.

**Comment by arfon** [comment]
@whedon remind @arfon in 6 weeks

**Comment by ysohn** [comment]
👋 @arfon  We deeply appreciate for the editor and the reviewers for their patience and apologize for the delay. We have completed the update and closed minor issues except 2 major issues raised by @martinmodrak. We'd very much appreciate for your further guidance.

**Comment by martinmodrak** [comment]
Hi, I did skim your responses and I think that (as I said before) the package being transparent about convergence (or lack of it) is what's important. I'll try to read and respond in depth, but I won't be able to have a response ready before Christmas break.

**Comment by arfon** [comment]
> Hi, I did skim your responses and I think that (as I said before) the package being transparent about convergence (or lack of it) is what's important. I'll try to read and respond in depth, but I won't be able to have a response ready before Christmas break.
 
Many thanks @martinmodrak. I'll check back in with you in the new year.

**Comment by martinmodrak** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02708/joss.02708/10.21105.joss.02708.pdf) :page_facing_up: :point_left:

**Comment by martinmodrak** [comment]
Sorry for the delay on my side. I finally got to review the changes made.

I still don't think a change to the underlying MCMC method is necessary/recommended for JOSS acceptance, but I think the current version of the package sets the users up for failure in several ways, neither of which was fixed by the update (which seems to have changed little beyond the JOSS paper itself):

1) The potential limitations of the MCMC method and limitations on identifiability and convergence of the model are mentioned only in the JOSS paper, but not in the documentation for `NetworkChange`, neither in the example vignette. 
2) It is hard for users to verify whether the MCMC has converged and whether the posterior is multimodal, making it impossible to check that the inferences they'll make from their datasets are valid. In particular:

    a) There should be a straightforward way to assess convergence of the relevant parameters. Rhat/ESS is an option I would recommend, but there definitely are others which the authors may choose to use. Having a simple way to show traceplots would also be nice (but not necessary).

    b) the `NetworkChange` method should by default issue a warning if any of the parameters has bad diagnostic values

    c)  Checking the diagnostics should be shown in the examples in the vignette.

    d) the `NetworkChange` method should accept a parameter to specify the number of chains to run, with default of at least 4. (suggestion: use the `future` package to run the individual chains, which will let you very easily allow users to run the chains in parallel). 

3) The default number of burnin and mcmc samples is very low. This means that non-convergence due to short warmup is more likely and only low precision for any variable (including the MCMC diagnostics) is achieved. Note that 100 samples give me quite low precision, even if the samples were completely uncorrelated (which they aren't). E.g.  if I take 100 i.i.d. samples from a normal distribution with mean 0 and sd 1, the observed mean will fluctuate between -0.2 and 0.2 , observed sd will be 0.86 - 1.13 and observed 0.975th quantile would be 1.44 - 2.38 (all are 95% intervals).

I think those are major shortcomings that need to be addressed. The ordering above is roughly the order of importance I see in those issues. I believe extending the documentation is easy and important and is IMHO a blocking issue for acceptance. Adding a way to access the diagnostics is also relatively easy and is IMHO a high priority, although not a 100% must. I feel less strongly about adding multiple chain support and changing the default sample numbers, so if there is a good reason not to do those, I am open to be convinced so.

Additionally, I think there is a misconception in the reply to https://github.com/jongheepark/NetworkChange/issues/6 as well as in the paper. I discuss that in my comment on the issue and expect to either be proven wrong or the wording in the paper to change accordingly.

**Comment by arfon** [comment]
I just emailed @ysohn to remind them we're awaiting their response still here.

**Comment by arfon** [comment]
@ysohn – I tried contacting you via email but the account associated with your JOSS submission appears to be disabled. If we don't hear back from you in the next couple of weeks that you're still interested in pursing this submission then we will reject due to inactivity.

**Comment by martinmodrak** [comment]
Maybe tagging @jongheepark as the repo owner could help? I also just noted that all the issues on the repository seem to have been deleted.

**Comment by ysohn** [comment]
@arfon @martinmodrak  We are terribly sorry for this. All issues are gone without our notice. We are trying to recover them, and let you know the progress.

**Comment by arfon** [comment]
@ysohn – thank you for the update. I have to say, the (presumably accidental) deletion of the reviewer feedback to date is very unfortunate. This submission has moved very slowly, and feedback to the reviewers has been very delayed. At this point I need you to do the following in order to save this submission from rejection:

(Assuming you're able to recover the issues somehow, perhaps from email?):

- Provide a detailed response to each and every issue flagged by the reviewers (@kakiac provided an excellent summary here: https://github.com/openjournals/joss-reviews/issues/2708#issuecomment-724284590)
- For each issue, link directly to the changes you've made _in code_ (or the paper) to make it very straightforward for the reviewers to assess these changes.

Please provide this feedback at the latest by 21st May 2022.

**Comment by arfon** [comment]
@ysohn – per my earlier message on the 21st April, I'm now proceeding to reject this submission. @akbaritabar, @marcjwilliams1, @martinmodrak, @kakiac – many thanks for your work here, and sorry this didn't work out.

**Comment by arfon** [comment]
@editorialbot reject
